snuba/subscriptions/subscription.py

```diff
@@ -22,14 +23,16 @@ class SubscriptionCreator:
 
     def __init__(self, dataset: Dataset):
         self.dataset = dataset
+        self.__partitioner = TopicSubscriptionDataPartitioner(
+            enforce_table_writer(dataset).get_stream_loader().get_default_topic_spec()
```

Comment:
We'll need to find a way to configure the topic partition count, as right now this more or less hardcoded to 1 for all topics. This is fine for local development but obviously won't work in production.

Reply:
Yeah I guess this is the first use case where we need to read the partition count for a topic (this was only used to create topics in bootstrap) clearly that assumption does not work anymore.
Would it make sense to load the topic metadata from kafka itself instead of relying on configuration ? We should be able to get the number of partitions for a topic.

Reply:
> Would it make sense to load the topic metadata from kafka itself instead of relying on configuration ? We should be able to get the number of partitions for a topic.

I think it would make sense to warn loudly if the configured value doesn't match the actual value but I'd be very hesitant to dynamically change this divisor.